downstairs/src/lib.rs

```diff
@@ -845,14 +845,39 @@ where
                         upstairs_id,
                         session_id,
                         gen,
+                        read_only,
+                        encrypted,
                     }) => {
                         if negotiated != 0 {
                             bail!("Received connect out of order {}",
                                 negotiated);
                         }
+
                         if version != 1 {
                             bail!("expected version 1, got {}", version);
                         }
+
+                        // Reject an Upstairs connection if there is a mismatch
+                        // of expectation
+                        {
+                            let ds = ads.lock().await;
+                            if ds.read_only != read_only {
+                                let mut fw = fw.lock().await;
+                                fw.send(Message::ReadOnlyMismatch {
```

Comment:
I think we want to bail here maybe?  Or send back the response, then bail?
If encrypted or read_only or version don't match what this downstairs expects, then there is really nothing more this downstairs can do with this connection to the upstairs.  Does that sound right?

Reply:
It does sound right - I switched to send response then bail in 9ec0a50